tests/test_migration_steps.py

```diff
@@ -0,0 +1,176 @@
+import shutil
+from pathlib import Path
+
+from tests import utils
+
+files_dir = Path(__file__).parent.parent / "files"
+# Necessary to be loaded into the PYTHONPATH so that migrations.py can be loaded successfully
+migration_steps = utils.load_module("migration_steps", files_dir / "migration_steps.py")
+
+
+def test_path_snapshot_file(tmp_file):
+    tmp_path = tmp_file.parent / "different-location"
+
+    migration_steps.path_snapshot(tmp_file, tmp_path)
+
+    assert (tmp_path / tmp_file.name).exists()
+
+
+def test_path_snapshot_folder(tmp_folder):
+    tmp_path = tmp_folder.parent / "different-location"
+
+    migration_steps.path_snapshot(tmp_folder, tmp_path)
+
+    assert (tmp_path / tmp_folder.name).exists()
+
+
+def test_path_validate(tmp_folder):
+    example = tmp_folder / "example-1.txt"
+    assert migration_steps.path_validate(example, True)
+    assert migration_steps.path_validate(example, None)
+    example.unlink()
+    assert migration_steps.path_validate(example, None)
+    assert migration_steps.path_validate(example, False)
+    assert migration_steps.path_validate(tmp_folder, True)
+    assert migration_steps.path_validate(tmp_folder, None)
+    shutil.rmtree(tmp_folder)
+    assert migration_steps.path_validate(tmp_folder, False)
+
+
+def test_path_rollback_file(tmp_file):
+    tmp_path = tmp_file.parent / "different-location"
+
+    contents = tmp_file.open().read()
+    migration_steps.path_snapshot(tmp_file, tmp_path)
+    tmp_file.unlink()
+    assert not tmp_file.exists()
+
+    migration_steps.path_rollback(tmp_file, tmp_path)
+    assert tmp_file.exists()
+    assert tmp_file.open().read() == contents
+
+
+def test_path_rollback_folder(tmp_folder):
+    tmp_path = tmp_folder.parent / "different-location"
+
+    original_contents = [fp.name for fp in tmp_folder.glob("*")]
```

Comment:
First, you can use `tmp_folder.iterdir()` instead of globbing everything :)

Second, you should be able to do normal equality on Path instances. So I think `original_contents = list(tmp_folder.iterdir())` and then the same below for `rolled_back_contents` and checking they're equal should work.